src/protocols/coda_transition_frontier.ml

```diff
@@ -160,6 +160,8 @@ module type Network_intf = sig
 
   type peer
 
+  type inet_addr
```

Comment:
Since we always substitute this for `Core.Unix.Inet_addr.t` let's not make it a variable

Reply:
`peer` is always `Network_peer.Peer.t` (I think), should that not be an opaque type?